src/components/shared/Popover.js

```diff
@@ -50,12 +53,24 @@ class Popover extends Component {
     const left = targetLeft + targetWidth / 4;
     const top = targetTop - height;
 
-    return { left, top };
+    return { left, top, dir: "up" };
   }
 
   renderPopover() {
     const { children, onMouseLeave } = this.props;
-    const { top, left } = this.state;
+    const { top, left, dir } = this.state;
+
+    if (dir === "up") {
+      return dom.div(
+        {
+          className: "popover popover-up",
```

Comment:
we can use the classnames helper to do `classnames("popover", { up: dir =="up" })`